src/Tensor.jl

```diff
@@ -53,8 +53,24 @@ Base.isequal(a::Tensor, b::Tensor) = allequal(labels.((a, b))) && allequal(paren
 
 labels(t::Tensor) = t.labels
 
-Base.replace(t::Tensor, old_new::Pair{Symbol,Symbol}...) =
-    Tensor(parent(t), replace(labels(t), old_new...); copy(t.meta)...)
+# NOTE: `replace` does not currenly support cyclic replacements
+function Base.replace(t::Tensor, old_new::Pair{Symbol,Symbol}...)
+    new_labels = replace(labels(t), old_new...)
+    new_meta = deepcopy(t.meta)
+
+    if haskey(new_meta, :alias)
+        full_mapping = Dict{Symbol, Symbol}(old_new)
+
+        updated_aliases = Dict{Symbol, Symbol}()
+        for (key, value) in new_meta[:alias]
+            updated_value = value in keys(full_mapping) ? full_mapping[value] : value
+            updated_aliases[key] = updated_value
+        end
+        new_meta[:alias] = updated_aliases
+    end
```

Comment:
Your approach although I must say is very nice, it doesn't work here since you update the `values(new_meta[:alias])`, not the `new_meta` `Dict`. And `map!` does not support `Dict`s. 

However, inspired from your code I managed to reduce it to:
```julia
    if haskey(new_meta, :alias)
        old_new_dict = Dict{Symbol, Symbol}(old_new)
        new_meta[:alias] = Dict(k => get(old_new_dict, v, v) for (k, v) in new_meta[:alias])
    end
```

Reply:
> However, inspired from your code I managed to reduce it to:
> 
> ```julia
>     if haskey(new_meta, :alias)
>         old_new_dict = Dict{Symbol, Symbol}(old_new)
>         new_meta[:alias] = Dict(k => get(old_new_dict, v, v) for (k, v) in new_meta[:alias])
>     end
> ```

Just a tip about your code. As `old_new` is already a `Base.Pairs`, you don't need to convert it into a `Dict`. Several `Dict`-methods are implemented for `Pairs` (which in the end is an immutable iterator of a `Dict`). Thus you can call `get` without converting it into a `Dict`.

```julia
    if haskey(new_meta, :alias)
        new_meta[:alias] = Dict(k => get(old_new, v, v) for (k, v) in new_meta[:alias])
    end
```

Still look at my `map!` implementation because it would avoid the allocation of a new `Dict`.

Reply:
I revised your implementation and you are right that the fact that it does not work is not because of the `map!` (sorry), but rather that when you do `i ∈ keys(old_new) ? old_new[i] : i` you are trying to access the value of `old_new` as if it were a dictionary, but it is a tuple of `Pair` objects. We can work around this problem if we convert it to a `Dict`:
```julia
    old_new_dict = Dict{Symbol, Symbol}(old_new)

    haskey(new_meta, :alias) && map!(values(new_meta[:alias])) do i
        get(old_new_dict, i, i)
    end
```
And now this works.

Reply:
Okay, i see. I found out that using `old_new_dict = pairs(NamedTuple(old_new))` works but in that case, it is equivalent to using an `ImmutableDict`.